ui/tsconfig.base.json

```diff
@@ -11,8 +11,7 @@
     "isolatedModules": true,
     "noUncheckedIndexedAccess": true,
     "composite": true,
-    "paths": {
-      "@perses-ui/*": ["./*/src"]
```

Comment:
the unit tests are failing. I'm wondering if it's related to this change.

I'm surprised we don't need it because when I worked around this topic and the possibility to run `npm start`, this part was the key to make it works.

Reply:
Oh man, this sent me down a rabbit hole. 😂 

OK, so you're right, this change _is_ why the tests are failing to compile. But with Typescript project references, this setting isn't actually supposed to be necessary. With project references, the `references` config in `tsconfig.json` points to the other projects that a given package depends on. Then when you run `tsc --build` (you have to use `--build` to get project references to compile correctly), the Typescript compiler takes into account the `references` config and builds those projects first.

The problem is that `jest` and `ts-jest` in particular doesn't currently deal with project references. Since it's a jest transformer that runs "on-demand" for individual files, it doesn't actually have a way to use the `--build` flag that would take care of building project references first. There's an [open issue in ts-jest](https://github.com/kulshekhar/ts-jest/issues/1648) currently where they're discussing how to deal with project references and has some workarounds.

I'm going to think about this a bit more, but I think for now, I may just use one of the workarounds (e.g. `moduleNameMapper`) suggested in that issue...

Reply:
I ended up bringing this config back in a `tsconfig.test.json` for running unit tests. We'll see what `ts-jest` ends up ultimately doing to better support project references.